lib/spack/spack/package.py

```diff
@@ -1346,6 +1349,40 @@ def build_process(input_stream):
             if not keep_prefix:
                 self.remove_prefix()
 
+    def repair_partial(self, reuse_prefix=False, reuse_stage=True):
```

Comment:
Can we rename this method to something more descriptive? Maybe "prepare_prefix_and_stage" or something like that? Also maybe rename "reuse_prefix" to "keep_prefix" and "reuse_stage" to "restage" (with the latter involving a value flip) so it's obvious to maintainers that this is managing the semantics of those two variables?